### PR TITLE
Ignore `flutter analyze` errors via analysis options.

### PR DESCRIPTION
**Pull request by pq**
Migrates error suppression from regexps to analysis options.
- Ignore TODOs.
- Ignore strong mode field and method overrides (see `InvalidFieldOverride` and `InvalidMethodOverride` in `strong/info.dart`.
- Remove stale regexp to capture refs to dead `dart:ui_internals` URI.
- Remove stale regexp to capture dead strong mode "will need runtime check to cast to type" error (https://github.com/dart-lang/sdk/issues/24542).

5 regexp downs, another small handful (~4) to go! :)

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 1]
```diff
@@ -313,14 +320,9 @@ linter:
...
       new RegExp('^\\[error\\] The argument type \'List<T>\' cannot be assigned to the parameter type \'List<.+>\''), // until we have generic methods, there's not much choice if you want to use map()
-      new RegExp(r'^\[error\] Field declaration .+ cannot be overridden in .+\.'), // we allow overriding fields (if they use super, ideally...)
-      new RegExp(r'^\[warning\] .+ will need runtime check to cast to type .+'), // https://github.com/dart-lang/sdk/issues/24542
       new RegExp(r'^\[error\] Type check failed: .*\(dynamic\) is not of type'), // allow unchecked casts from dynamic
-      new RegExp('^\\[error\\] Target of URI does not exist: \'dart:ui_internals\''), // https://github.com/flutter/flutter/issues/83
-      new RegExp(r'\[info\] TODO.+'),
-      new RegExp('\\[warning\\] Missing concrete implementation of \'RenderObject\\.applyPaintTransform\''), // https://github.com/dart-lang/sdk/issues/25232
+      //new RegExp('\\[warning\\] Missing concrete implementation of \'RenderObject\\.applyPaintTransform\''), // https://github.com/dart-lang/sdk/issues/25232
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: Did you mean to leave this commented out?

**Comment by eseidelGoogle** [comment]
This looks like a review for @Hixie

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 1 above)
Comment (pq) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: No!  Thanks for the catch.  Fixing.

**Comment by pq** [comment]
Assuming travis is green, @Hixie PTAL.

**Comment by Hixie** [comment]
lgtm

**Pull request merged**